### PR TITLE
Initial refactor of the developers-guide

### PR DESCRIPTION
**Pull request by offtherailz**
Here a small refactor of the developer guide. 
The main improvements are :
 - more rational organization of the contents
 - added code conventions
 - UI guidelines now can be added here

Please comment or contribute. If the contributions are too much we can do a branch and work on that together.

### CODE REVIEW

**Comment by Gnafu** [comment]
The textboxes suggestion text casing should be consistent, I suggest Uppercase.

**Comment by offtherailz** [comment]
We should add  textboxes suggestion text and other details in the style guidelines.

File: docs/developer-guide/code-conventions.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+It will help also future refactoring because any change to the state structure (from the reducer point of view) or data source (from the components point of view) requires only changes to the interested selectors.
+
+A selector should be placed into the proper `selectors/<state-slice>.js` file with the same name of the relative reducer.
+When a selector retrieves data from more than one state slices, you should place it in the selector nearest by concern. For instance `isFeatureGridOpen` should be placed into `featuregrid`
+
+If you don't work on a core functionality, where the state is shared between many components, defining the selector directly in the plug-in is not denied (e.g. examples).
+
+## Prefer `initialState` over plugin configuration
+In the past we had a plug-in configuration and initial state. The plug-in configuration is now considered a bad practice. Use the `initialState` to configure the application.
```
Comment (mbarto) [inline docs/developer-guide/code-conventions.md]: I don't agree, you cannot put everything in the initial state, only stuff that is connected through redux, configuration is still a good practice for properties that are not connected to the state (elsewhere, how can you configure them?)

Comment (offtherailz) [reply docs/developer-guide/code-conventions.md]: Good point. Can you specify that?

**Pull request merged**